### PR TITLE
getData formats data in a usable way, update_info implemented

### PR DESCRIPTION
**Pull request by busgurlu**

### CODE REVIEW

**Comment by mikehaertl** [comment]
@busgurlu This looks great, thanks!

I have some minor comments mainly about coding style. But before I go into depth I have one question: Are we sure, that we catch any possible fields in the output of `pdftk dump_data` or could there maybe be more fields? Did you find this documented somewhere?

If the parser finds a line it doesn't understand, what does it do?

**Comment by busgurlu** [comment]
I wasn't able to find any documentation on this. These fields are the ones I come up on the files I'm working on, as well as on Google. Works for me at the moment.

File: src/Pdf.php [hunk 1]
```diff
@@ -263,6 +263,27 @@ public function fillForm($data, $encoding = 'UTF-8', $dropXfa = true, $format =
...
+     */
+    public function updateInfo($data, $encoding = 'UTF-8')
+    {
+        $this->constrainSingleFile();
+        if (is_array($data)) {
+            $data = new \mikehaertl\pdftk\InfoFile($data, null, null, null, $encoding);
+        }
+        $this->getCommand()
+        ->setOperation($encoding == 'UTF-8' ? 'update_info_utf8' : 'update_info')
+        ->setOperationArgument($data, true);
```
Comment (mikehaertl) [inline src/Pdf.php]: These 2 lines should be indented.

File: src/Pdf.php [hunk 2]
```diff
@@ -263,6 +263,27 @@ public function fillForm($data, $encoding = 'UTF-8', $dropXfa = true, $format =
...
+     * @param string|array $data either a InfoFile filename or an array with 
+     * form field data (name => value)
+     * @param string the encoding of the data. Default is 'UTF-8'.
+     * @return \mikehaertl\pdftk\Pdf the pdf instance for method chaining
+     */
+    public function updateInfo($data, $encoding = 'UTF-8')
+    {
+        $this->constrainSingleFile();
+        if (is_array($data)) {
+            $data = new \mikehaertl\pdftk\InfoFile($data, null, null, null, $encoding);
```
Comment (mikehaertl) [inline src/Pdf.php]: Namespace should not be neccessary here, right?

**Comment by mikehaertl** [comment]
Ok, I understand. Finding documentation on PDF specs is really a pain. So let's keep thing as they are.

Regarding the coding style I nowadays try to follow [PSR-2](https://www.php-fig.org/psr/psr-2/). It would be cool if you could fix the style. You could use [php-cs-fixer](https://cs.symfony.com/) for it which seems to do a good job.

Main things I found for PSR-2:
 * Missing spaces before brackets: `if(...` instead of `if (`
 * `else` and `elseif` should not be on a new line
 * Missing spaces before/after `=>`

Apart from PSR-2 I also would prefer the following:
 * use `protected` instead of `private` to make extending classes easier
 * always use `===` when comparing strings
 * use spaces around `=`, `==` and `===` (I know, the current code doesn't follow this yet - will fix it soon)

If you don't find the time, I can also fix the above after the merge, if you don't mind.

File: src/InfoFields.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                    $buffer['Value'] = $value;
+                }
+                if(isset($buffer['Value']) && isset($buffer['Key'])) {
+                    $output['Info'][$buffer['Key']] = $buffer['Value'];
+                    $buffer = array();
+                    $expectType = null;
+                }
+                continue;
+            }
+            if(!is_null($expectType)){
```
Comment (mikehaertl) [inline src/InfoFields.php]: I would prefer `=== null`.

**Comment by mikehaertl** [comment]
> Sure, no problem.

Sorry, do you mean I should fix them (no problem for me)? Or will you?

**Comment by busgurlu** [comment]
I don't mind, please fix, change or discard as you like. I have to move on. Sorry.

**Comment by mikehaertl** [comment]
Great, I've just released 0.6.0 which includes your addon. Thanks again!

**Pull request merged**